browser_use/agent/service.py

```diff
@@ -1456,3 +1467,14 @@ async def _update_action_models_for_page(self, page) -> None:
 		# Update done action model too
 		self.DoneActionModel = self.controller.registry.create_action_model(include_actions=['done'], page=page)
 		self.DoneAgentOutput = AgentOutput.type_with_custom_actions(self.DoneActionModel)
+
+	def update_available_file_paths(self, file_paths: list[str]) -> None:
+		"""
+		Update the available file paths that the agent can access.
+
+		Args:
+			file_paths: List of file paths that the agent can access
+		"""
+		self.settings.available_file_paths = file_paths
+		self._message_manager.settings.available_file_paths = file_paths
```

Comment:
this overwrites both of these, are you sure you dont want to set-append?
```suggestion
		self.settings.available_file_paths = list({*self.settings.available_file_paths, *file_paths})
		self._message_manager.settings.available_file_paths = list({*self._message_manager.settings.available_file_paths, *file_paths})
```